audio-worklet/bitcrusher-test.html

```diff
@@ -16,171 +16,184 @@
-    	                  buffer.getChannelData(c)[i] + ' did not match ' +
-    	                  ' expectation, ' + last);
-    	          if (i % reduction === 0) last = buffer.getChannelData(c)[i];
-    	        }
-    	     } 
-    	  });  
-    	}
+      testSampleReduction();
+      testBitcrusherInAudioContext();
+      testMultipleChannels();
+      testIllegalInput();
+
+      /**
+       * Populate array with 0:99 and verify the bitcrusher holds an array
+       * where the number of consecutive equal numbers is controlled by the 
+       * variable factor. This verfies the sample rate reduction component
+       * of bitcrusher operating outside of an audio context.
+       */
+      function testSampleReduction() {
+        let context = new OfflineAudioContext(1, 44100, 44100);
+
+        for (let reduction = 1; reduction < 10; reduction++) {
+          const bufferSize = 5;
+          let referenceBuffer = new Float32Array(bufferSize);
+          let outputBuffer = new Float32Array(bufferSize);
+          let bitcrusher =
+              new Bitcrusher(context, {bitDepth: 24, reduction: reduction});
+
+          for (let i = 0; i < bufferSize; i++) {
+            referenceBuffer[i] = i;
+          }
+          bitcrusher.processBuffer_(referenceBuffer, outputBuffer, 0);
+          
+          // Verify computed values match expected values.
+          outputBuffer.map((data, index) => {
+            let expected = Math.floor(index / reduction) * reduction;
+            console.assert(
+                data === expected, 
+                'computed ' + data + ' but expected ' + expected);
+          });
+        }
+      }
+
+      /**
+       * Create one oscillator, and connect it to a bitcrusher with variables 
+       * set to avoid any effect and a delay node which compensates for the 
+       * script processor latency. Then verify that the samples are nearly
+       * identical.
+       */
+      function testBitcrusherInAudioContext() { 
+        let context = new OfflineAudioContext(2, 44100, 44100);
+        const reduction = 1;
+        const bitDepth = 24;
+        const bufferSize = 512;
+        const channels = 1;
+        let oscillator = new OscillatorNode(context);
+        let merger = new ChannelMergerNode(context, {numberOfInputs: 2});
+        let bitcrusher = new Bitcrusher(
+            context, {bufferSize, channels, bitDepth, reduction});
+        // Accomodate for script processor latency by delaying the oscillator.
+        let delay = new DelayNode(context, {
+            delayTime: (bufferSize / context.sampleRate)
+        });
+
+        oscillator.connect(bitcrusher.input);
+        bitcrusher.output.connect(merger, 0, 0);
+        oscillator.connect(delay).connect(merger, 0, 1);
+        merger.connect(context.destination);
+        oscillator.start();
+
+        // When audio buffer is ready, verify bitcrushed samples are unaltered.
```

Comment:
"unaltered" doesn't really match what you're doing, because the samples aren't actually unaltered.  Fix this comment.